tests/Unit/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+set(DAWG_DIR "Dawg")
+
+###############################################################################
+# AddUnitTest
+# A macro that will add a unit test to the testing queue
+# Example: AddUnitTest(dng::seq)
+#   - creates the `Unit.dng::seq` test
+#   - creates the `unittest_dng_seq` target from the file ./dng/seq.cc
+#   - builds the target and runs the right test test-command
+#   - the target is linked against Boost::UNIT_TEST_FRAMEWORK and libdng
+
+macro(AddUnitTest NAME)
+    string(REPLACE "::" "/" head "${NAME}")
+    string(REPLACE "::" "_" exe "${NAME}")
+    set(target "unittest_${exe}")
+    add_executable("${target}" EXCLUDE_FROM_ALL "${head}.cpp")
+    # TODO WIN64: add .exe if needed
+    set_target_properties("${target}" PROPERTIES OUTPUT_NAME "${head}")
+    get_filename_component(dir "${head}" DIRECTORY)
+    file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${dir}")
+
+    target_include_directories("${target}" BEFORE PRIVATE "${CMAKE_SOURCE_DIR}/src/include")
+
+    target_link_libraries("${target}"
+            libdawg2
+            ${Boost_FILESYSTEM_LIBRARY}
+            ${Boost_SYSTEM_LIBRARY}
+            ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+            testu01
```

Comment:
Yeah, we need to create one if it doesn't exist.

--
Reed A. Cartwright, PhD
Barrett Honors Faculty
Assistant Professor of Genomics, Evolution, and Bioinformatics
School of Life Sciences
Human and Comparative Genomics Laboratory
The Biodesign Institute
Arizona State University
==================
Availability: http://links.asu.edu/CartwrightCalendar
Address: The Biodesign Institute, PO Box 875301, Tempe, AZ 85287-5301 USA
Packages: The Biodesign Institute, 1001 S. McAllister Ave, Tempe, AZ
85287-5301 USA
Office: Biodesign A-224A, 1-480-965-9949
Website: http://cartwrig.ht/

On Mon, Sep 12, 2016 at 1:17 PM, Zach <notifications@github.com> wrote:

> In tests/Unit/CMakeLists.txt
> <https://github.com/reedacartwright/dawg/pull/12#discussion_r78446220>:
>
> > +    string(REPLACE "::" "_" exe "${NAME}")
> > +    set(target "unittest_${exe}")
> > +    add_executable("${target}" EXCLUDE_FROM_ALL "${head}.cpp")
> > +    # TODO WIN64: add .exe if needed
> > +    set_target_properties("${target}" PROPERTIES OUTPUT_NAME "${head}")
> > +    get_filename_component(dir "${head}" DIRECTORY)
> > +    file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${dir}")
> > +
> > +    target_include_directories("${target}" BEFORE PRIVATE "${CMAKE_SOURCE_DIR}/src/include")
> > +
> > +    target_link_libraries("${target}"
> > +            libdawg2
> > +            ${Boost_FILESYSTEM_LIBRARY}
> > +            ${Boost_SYSTEM_LIBRARY}
> > +            ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
> > +            testu01
>
> Should there be a Find*.cmake to find the libtestu01 library?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/reedacartwright/dawg/pull/12/files/15697e935de1b84fb839564505829f67d0b6bcd8#r78446220>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGOHvqTl1wtWxYeA-eKpSa0PswjTn7dks5qpbNMgaJpZM4JdPuB>
> .
>